### PR TITLE
[SANS913] Resolve import plotSpectrum error in Mantid package

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

- original ORNL tracker: [SANS913](https://code.ornl.gov/sns-hfir-scse/sans/sans-backend/-/issues/913)

tl;dr;
-----

This PR makes the import of `plotSpectrum` optional so that it does not break the following statement in `mantid` package (without qt stuff).

```python
from mantid.simpleapi import *
```

**To test:**

- create a local conda package using the following command

```bash
$MANTID_ROOT/buildconfig/Jenkins/Conda/package-conda $MANTID_ROOT --build-mantid --build-current-branch
```

- install the tarball created under `mantid/conda-bld/linux-64` in a fresh environment without previous mantid

- start Python interpreter and perform the following:

```
from mantid.simpleapi import *
```

This should only generate a warning, and no error.

> On Ubuntu 22.04, the local build will add qt-5.12 as part of the dependency, but no `mantidworkbench` is built.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Can you mention in the warning that `plot_reduction` will not work as a result?

Also, `plot_reduction` should give a better error than unknown variable/symbol/whatever it will currently say. A way to do it is to change the conditional import to be
```python
try:
    import plotSpectrum
except ImportError:
    logger.warning("plotSpectrum not available")
    plotSpectrum = None
```
Then `plot_reduction` can check `if plotSpectrum is None` and throw an exception with a meaningful message

**Comment by KedoKudo** [comment]
> Can you mention in the warning that `plot_reduction` will not work as a result?
> 
> Also, `plot_reduction` should give a better error than unknown variable/symbol/whatever it will currently say. A way to do it is to change the conditional import to be
> 
> ```python
> try:
>     import plotSpectrum
> except ImportError:
>     logger.warning("plotSpectrum not available")
>     plotSpectrum = None
> ```
> 
> Then `plot_reduction` can check `if plotSpectrum is None` and throw an exception with a meaningful message

The error messages are updated as suggested.

**Review by peterfpeterson — APPROVED** [review]

**Comment by peterfpeterson** [comment]
I ran the following test

First install using
```sh
conda  create -n testing  ./mantid-6.5.20230109.1306-py38hefa2ac8_0.tar.bz2
conda activate testing
conda install python=3.8  # this being required as a separate step feels strange
```

Then 
```sh
$ python                                                                                                                                
Python 3.8.15 | packaged by conda-forge | (default, Nov 22 2022, 08:46:39)                                                                                                 
[GCC 10.4.0] on linux                                                                                                                                                      
Type "help", "copyright", "credits" or "license" for more information.                                                                                                     
>>> from mantid import __version__                                                                                                                                         
>>> __version__
'6.5.20230109.1306'
>>> from mantid.simpleapi import *
FrameworkManager-[Notice] Welcome to Mantid 6.5.20230109.1306
FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid
DownloadInstrument-[Notice] All instrument definitions up to date
Python-[Warning] plotSpectrum not available
```

File: scripts/Inelastic/IndirectReductionCommon.py [hunk 1]
```diff
@@ -4,10 +4,16 @@
 #   NScD Oak Ridge National Laboratory, European Spallation Source,
 #   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 # SPDX - License - Identifier: GPL - 3.0 +
-from mantid.simpleapi import AppendSpectra, DeleteWorkspace, Load, plotSpectrum
+from mantid.simpleapi import AppendSpectra, DeleteWorkspace, Load
 from mantid.api import AnalysisDataService, WorkspaceGroup, AlgorithmManager
 from mantid import mtd, logger, config
 
+try:
+    import plotSpectrum
```
Comment (martyngigg) [inline scripts/Inelastic/IndirectReductionCommon.py:12]: I didn't spot this but I think this import is wrong? Should it not be `from mantid.simpleapi import plotSpectrum`?

It looks like if you start Workbench at the moment you get: `plotSpectrum not available` printed.

Comment (martyngigg) [reply scripts/Inelastic/IndirectReductionCommon.py:12]: I've created #34929 to track it. Would it be possible to get this fixed as a high priority before we put the beta out?

**Comment by MialLewis** [comment]
I've got a user seeing this issue importing `mantid` `6.6.0`?

`FrameworkManager-[Notice] Welcome to Mantid 6.6.0 FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid6.6
DownloadInstrument-[Notice] Downloading 11 files from the instrument repository DownloadInstrument-[Notice] A new Facilities.xml file has been downloaded, this will take effect next time Mantid is started.
Python-[Warning] plotSpectrum not available`

**Comment by MialLewis** [comment]
Do we want a warning to be printed as default for all those just using mantid as a python package, it could be (has been) confusing. One for @gemmaguest.
